endpoints/openrtb2/auction.go

```diff
@@ -1354,20 +1358,35 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 
 	// Apply the Stored BidRequest, if it exists
 	resolvedRequest := requestJson
-	if hasStoredBidRequest {
-		resolvedRequest, err = jsonpatch.MergePatch(storedRequests[storedBidRequestId], requestJson)
+
+	if deps.cfg.GenerateRequestID {
+		isAppRequest, err := checkIfAppRequest(requestJson)
 		if err != nil {
-			hasErr, Err := getJsonSyntaxError(requestJson)
-			if hasErr {
-				err = fmt.Errorf("Invalid JSON in Incoming Request: %s", Err)
-			} else {
-				hasErr, Err = getJsonSyntaxError(storedRequests[storedBidRequestId])
-				if hasErr {
-					err = fmt.Errorf("Invalid JSON in Stored Request with ID %s: %s", storedBidRequestId, Err)
-					err = fmt.Errorf("ext.prebid.storedrequest.id refers to Stored Request %s which contains Invalid JSON: %s", storedBidRequestId, Err)
+			return nil, nil, []error{err}
+		}
+		if isAppRequest {
+			uuidPatch, err := generateUuidForBidRequest(deps.uuidGenerator)
```

Comment:
The `uuidPatch` is being generated for app requests, but it should only be generated when also `hasStoredBidRequest` is true.

Reply:
Why do we want to disable generating request IDs when there is no stored request for an App request?

Reply:
Talked to SyntaxNode and this is desired behavior. Should be documented somewhere, in the end (not in the code, but the docs elsewhere) since it is not intuitively obvious when this feature should or should not kick in.